includes/class-woocommerce-delivery-notes.php

```diff
@@ -609,6 +530,87 @@ public function is_woocommerce_activated() {
 				return false;
 			}
 		}
+		/**
+		 * It will delete the tracking option from the database.
+		 */
+		public function ts_reset_tracking_setting() {
+
+			if ( isset( $_GET ['ts_action'] ) && 'reset_tracking' === $_GET ['ts_action'] ) { //phpcs:ignore
+				Tyche_Plugin_Tracking::reset_tracker_setting( 'wcdn' );
+				$ts_url = remove_query_arg( 'ts_action' );
+				wp_safe_redirect( $ts_url );
+			}
+		}
+		/**
+		 * Add tracker completed.
+		 */
+		public function init_tracker_completed() {
+			header( 'Location: ' . admin_url( 'admin.php?page=wc-settings&tab=wcdn-settings&setting=wcdn_general' ) );
+			exit;
+		}
+
+		/**
+		 * Display admin notice on specific page.
+		 *
+		 * @param array $is_flag Is Flag defailt value true.
+		 */
+		public function wcdn_ts_tracker_display_notice( $is_flag ) {
+			global $current_section;
+			if ( isset( $_GET['page'] ) && 'wc-settings' === $_GET['page'] ) { // phpcs:ignore
+				$is_flag = true;
+			}
+			return $is_flag;
+		}
+
+		/**
+		 * Send the plugin data when the user has opted in
+		 *
+		 * @hook ts_tracker_data
+		 * @param array $data All data to send to server.
+		 *
+		 * @return array $plugin_data All data to send to server.
+		 */
+		public function wcdn_ts_add_plugin_tracking_data( $data ) {
+			$plugin_short_name = 'wcdn';
+			if ( ! isset( $_GET[ $plugin_short_name . '_tracker_nonce' ] ) ) {
+				return $data;
+			}
+
+			$tracker_option = isset( $_GET[ $plugin_short_name . '_tracker_optin' ] ) ? $plugin_short_name . '_tracker_optin' : ( isset( $_GET[ $plugin_short_name . '_tracker_optout' ] ) ? $plugin_short_name . '_tracker_optout' : '' ); // phpcs:ignore
+			if ( '' === $tracker_option || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET[ $plugin_short_name . '_tracker_nonce' ] ) ), $tracker_option ) ) {
+				return $data;
+			}
+
+			$data = self::wcdn_add_plugin_tracking_data( $data );
+			return $data;
+		}
+
+		/**
+		 * Plugin's data to be tracked when Allow option is choosed.
+		 *
+		 * @hook ts_tracker_data
+		 *
+		 * @param array $data Contains the data to be tracked.
+		 *
+		 * @return array Plugin's data to track.
+		 */
+		public function wcdn_add_plugin_tracking_data( $data ) {
```

Comment:
@nidhitatosaniya If this function has not been used more than once, then please collapse and remove it.